tests/e2e/APNSNetworkServiceTest.php

```diff
@@ -0,0 +1,54 @@
+<?php
+declare( strict_types = 1 );
+
+class APNSNetworkServiceIntegrationTest extends APNSTest {
+
+	function setUp(): void {
+		// Ensure the server is running before each test
+		$this->assertTrue( $this->reset_mock_server() );
```

Comment:
```suggestion
		$this->assertTrue( $this->reset_mock_server(), "The mock server doesn't appear to be running." );
```

I intentionally run the tests without spinning up the mock server to see what happened. The error message was just about the failed `assertTrue`, but it could have been more informative.

Reply:
Great point!